src/OmniSharp.Script/ScriptProjectSystem.cs

```diff
@@ -67,48 +57,66 @@ public void Initalize(IConfiguration configuration)
             Context.RootPath = Env.Path;
             Logger.LogInformation($"Found {allCsxFiles.Length} CSX files.");
 
-            // TODO: write and adapter to implement the new ScriptCs ILogProvider interface
-            #pragma warning disable 0618
-            //script name is added here as a fake one (dir path not even a real file); this is OK though -> it forces MEF initialization
-            var baseScriptServicesBuilder = new ScriptServicesBuilder(new ScriptConsole(), LogManager.GetCurrentClassLogger())
-                    .LogLevel(LogLevel.Debug)
-                    .Cache(false)
-                    .Repl(false)
-                    .ScriptName(Env.Path)
-                    .ScriptEngine<NullScriptEngine>();
+            // explicitly inherit scripting library references to all global script object (InteractiveScriptGlobals) to be recognized
+            var inheritedCompileLibraries = DependencyContext.Default.CompileLibraries.Where(x =>
+                    x.Name.ToLowerInvariant().StartsWith("microsoft.codeanalysis")).ToList();
 
-            var scriptServices = baseScriptServicesBuilder.Build();
+            var runtimeContexts = File.Exists(Path.Combine(Env.Path, "project.json")) ? ProjectContext.CreateContextForEachTarget(Env.Path) : null;
```

Comment:
If I understand what this is doing, it'll probably need to change in the future for csproj .NET Core. I might not understand though. :smile: